config-linux.md

```diff
@@ -35,39 +34,29 @@ The following parameters can be specified to setup namespaces:
     * **`user`** the container will be able to remap user and group IDs from the host to local users and groups within the container.
     * **`cgroup`** the container will have an isolated view of the cgroup hierarchy.
 
-* **`path`** *(string, OPTIONAL)* - path to namespace file in the [runtime mount namespace](glossary.md#runtime-namespace)
+    Values have the following properties:
+
+    * **`path`** *(string, OPTIONAL)* - path to namespace file in the [runtime mount namespace](glossary.md#runtime-namespace)
 
 If a path is specified, that particular file is used to join that type of namespace.
-If a namespace type is not specified in the `namespaces` array, the container MUST inherit the [runtime namespace](glossary.md#runtime-namespace) of that type.
+If a namespace type is not specified in the `namespaces` object, the container MUST inherit the [runtime namespace](glossary.md#runtime-namespace) of that type.
 If a new namespace is not created (because the namespace type is not listed, or because it is listed with a `path`), runtimes MUST assume that the setup for that namespace has already been done and error out if the config specifies anything else related to that namespace.
 
 ###### Example
 
 ```json
-    "namespaces": [
-        {
-            "type": "pid",
+    "namespaces": {
+        "pid": {
             "path": "/proc/1234/ns/pid"
         },
-        {
-            "type": "network",
+        "network": {
             "path": "/var/run/netns/neta"
         },
-        {
-            "type": "mount"
-        },
-        {
-            "type": "ipc"
-        },
-        {
-            "type": "uts"
-        },
-        {
-            "type": "user"
-        },
-        {
-            "type": "cgroup"
-        }
+        "mount": {},
+        "ipc": {},
+        "uts": {},
+        "user": {},
+        "cgroup": {}
```

Comment:
I'm ok making that distinction (like I'm ok [distinguishing unset and `null`][1]).  If the consensus is that we shouldn't make that distinction, alternatives include:

a. Using `anyOf` to allow boolean or namespace-object values and requiring `path` in namespace-objects.  E.g. `"pid": true` or `"pid": {"path": "..."}`.  This is clear in JSON Schema, but may be awkward to unmarshal in Go.

b. Replace namespace objects with the path string, and distinguish between unset, empty, and non-empty strings.  This makes namespace payload extension more difficult, but I don't hear anyone calling for namespace extension at the moment.

c. Keep the namespace object, require `path`, and distinguish between empty and non-empty strings.  This is just exchanging empty-string checking (one level deeper) for the empy-object check that currently concerns you.

[1]: https://groups.google.com/a/opencontainers.org/forum/m/#!topic/dev/nfuv8NnOsUg